sympy/core/tests/test_numbers.py

```diff
@@ -121,6 +121,13 @@ def test_mod():
     assert Integer(10) % 4 == Integer(2)
     assert 15 % Integer(4) == Integer(3)
 
+    h = Symbol('h')
+    m = h ** 2 % h
+    k = h ** -2 % h
+
+    assert m == h * (h % 1)
+    assert k != 0
```

Comment:
These are always tricky because there are many wrong answers for k that would pass this test. A more positive way around this is to test k against the unevaluated form: `assert k == Mod(h**-2, h, evaluate=False)`

Reply:
Don't forget to import Mod at the top of test_numbers.py or else you will waste a test cycle ;-)